pandas/tests/io/parser/test_common.py

```diff
@@ -2040,6 +2040,14 @@ def test_read_csv_memory_growth_chunksize(all_parsers):
             pass
 
 
+def test_read_csv_raises_on_header_prefix():
```

Comment:
can you use the all_parsers fixture here

Reply:
I first tried to do that in c053a8f but looks likey `parsers.read_csv` is not going to `CParserWrapper`. Which is what we want to test.

Reply:
sure it is, what exactly was failing? this is the standard pattern for parser tests.

Reply:
The test is to check whether the error is being raised or not, and the error is raised from `CParserWrapper` So, the whole test fails here.

Reply:
You need to apply your fix to all engines.  You just applied to `CParserWrapper`.

https://github.com/pandas-dev/pandas/commit/c053a8ff631db81035fc8a27a668b6e87ed8cf43 is the correct test.